ui/src/app/shared/utils.ts

```diff
@@ -6,3 +6,16 @@ export function hashCode(str: string) {
     }
     return hash;
 }
+
+// concatMaps merges two maps. Later args take precedence where there's a key conflict.
+export function concatMaps(...maps: (Map<string, string> | null)[]): Map<string, string> {
```

Comment:
Javascript already has a "built-in" way to merge maps:

```
var merged = new Map([...map1, ...map2, ...map3])
```

No a big deal though. Please feel free to keep `concatMaps` if you like it more

Reply:
I feel like I went with this to solve some kind of type check issue.... idk, I'll take another look at it. Thanks!  :-)